ostd/src/task/task/mod.rs

```diff
@@ -320,7 +325,10 @@ impl TaskOptions {
         // have any arguments, so we only need to align the stack pointer to 16 bytes.
         ctx.set_stack_pointer(crate::mm::paddr_to_vaddr(new_task.kstack.end_paddr() - 16));
 
-        Ok(Arc::new(new_task))
+        let new_task = Arc::new(new_task);
+        scheduler::enqueue(new_task.clone());
```

Comment:
Again, the name `enqueue` is not accurate. A task may be enqueued because it is spawn or waken. I suggest renaming this method to `run_new_task`.